### PR TITLE
Clean up scrape config duplicate targets

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR will remove duplicate scrape config target to clean up the prometheus meta operator scrape config.

Removed targets:
- cluster-autoscaler: Labels were added in version [1.7.4](https://github.com/giantswarm/cluster-autoscaler-app/compare/v1.17.3...v1.17.4) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22cluster-autoscaler-app%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22cluster-autoscaler-app%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- chart-operator: Labels were added in version [2.3.1](https://github.com/giantswarm/chart-operator/compare/v2.3.0...v2.3.1) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22chart-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22chart-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
-  net-exporter: Labels were added in version [1.9.1](https://github.com/giantswarm/net-exporter/compare/v1.9.0...v1.9.1) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22net-exporter%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22net-exporter%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- kube-state-metrics-app: Labels were added in version [1.2.0](https://github.com/giantswarm/kube-state-metrics-app/compare/v1.1.1...v1.2.0) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22kube-state-metrics-app%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22kube-state-metrics-app%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- etcd-backup-operator:  Labels were added in version [1.2.0](https://github.com/giantswarm/etcd-backup-operator/compare/v1.1.0...v1.2.0) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22etcd-backup-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22etcd-backup-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- node operator: Labels were added in version [1.1.0](https://github.com/giantswarm/node-operator/compare/v1.0.1...v1.1.0) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22node-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22node-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- release operator: Labels were added in version [2.1.0](https://github.com/giantswarm/release-operator/compare/v2.0.0...v2.1.0) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22release-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22release-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- organization-operator: Labels were added in version [0.5.0](https://github.com/giantswarm/organization-operator/compare/v0.4.0...v0.5.0) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22organization-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22organization-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- prometheus-meta-operator: Labels were added in version [1.2.0](https://github.com/giantswarm/prometheus-meta-operator/compare/v1.1.0...v1.2.0) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22prometheus-meta-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22prometheus-meta-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- rbac-operator: Labels were added in version [0.6.0](https://github.com/giantswarm/rbac-operator/compare/v0.5.0...v0.6.0) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22rbac-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22rbac-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- azure-operator: Labels were added in version [5.0.0-alpha1](https://github.com/giantswarm/azure-operator/compare/v4.2.0...v5.0.0-alpha1) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22azure-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22azure-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- kvm-operator: Labels were added in version [3.12.2](https://github.com/giantswarm/kvm-operator/compare/v3.12.1...v3.12.2) and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22kvm-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22kvm-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
- grafana: Unneeded as we enabled the `service monitor`
- azure-collector: Unneeded as we enabled the `service monitor` https://github.com/giantswarm/azure-collector/pull/258
- app-operator: Labels were added in version 2.2.0 and versions below this one [are not installed](https://giantswarm.grafana.net/explore?left=%7B%22datasource%22:%22Cortex%22,%22queries%22:%5B%7B%22expr%22:%22sum(aggregation:giantswarm:app_deployed_workload_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22app-operator%5C%22%7D)%20by%20(version)%5Cnor%20sum(aggregation:giantswarm:app_deployed_management_cluster_total%7Bcustomer%3D~%5C%22.*%5C%22,provider%3D~%5C%22.*%5C%22,pipeline%3D%5C%22stable%5C%22,%20app%3D%5C%22app-operator%5C%22%7D)%20by%20(version)%22,%22format%22:%22time_series%22,%22instant%22:false,%22interval%22:%22%22,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000006%22%7D%7D%5D,%22range%22:%7B%22from%22:%22now-5m%22,%22to%22:%22now%22%7D%7D&orgId=1) in any cluster
 
## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

**Comment by hervenicol** [comment]
Wow, that's a lot of work!

TLDR, but I'd like to understand better:
* this PR removes labels rewriting, because the labels are now set from the source?
* this means it is not supposed to change anything in the stored data, nor on the dashboards?
* meaning this simplifies prometheus config files without impacting the data?

**Comment by QuentinBisson** [comment]
> this PR removes labels rewriting, because the labels are now set from the source?
This PR removes unneeded scrape config because the targets are scraped from other places, either a service monitor or this scrape config: https://github.com/giantswarm/prometheus-meta-operator/blob/4cb35b9fb565d6275411a14cab9901b4d1297595/files/templates/scrapeconfigs/additional-scrape-configs.template.yaml#L615
 
> this means it is not supposed to change anything in the stored data, nor on the dashboards? meaning this simplifies prometheus config files without impacting the data?

Yes you are right because we still scrape the same targets :)

**Review by hervenicol — APPROVED** [review]
Huge and tedious work, so TLDR, but looks like you've carefully tracked everything.
So YOLO, LGTM ;)
